packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -231,7 +231,15 @@ Future<String?> _getCodeSigningIdentityDevelopmentTeam({
 
   final Process opensslProcess = await processUtils.start(
     const <String>['openssl', 'x509', '-subject']);
-  await (opensslProcess.stdin..write(signingCertificateStdout)).close();
+
+  await ProcessUtils.writeToStdinGuarded(
+    stdin: opensslProcess.stdin,
+    content: signingCertificateStdout,
+    onError: (Object? error, _) {
+      throw Exception('Unexpected error when writing to openssl: $error');
+    },
```

Comment:
If this ends up becoming a pattern (it very much probably will), we may want to consider creating like a `writeToStdinUnsafe` method that accepts something like a `message` or `taskDescription`. This new method would then throw a custom exception type from that.